src/snowflake/snowpark/modin/pandas/series.py

```diff
@@ -141,16 +141,18 @@ def __init__(
             if name is None:
                 name = MODIN_UNNAMED_SERIES_LABEL
                 if (
-                    isinstance(data, (pandas.Series, pandas.Index))
+                    isinstance(data, (pandas.Series, pandas.Index, pd.Index))
                     and data.name is not None
                 ):
                     name = data.name
 
             query_compiler = from_pandas(
                 pandas.DataFrame(
                     pandas.Series(
-                        data=data,
-                        index=index,
+                        data=data.to_pandas() if isinstance(data, pd.Index) else data,
```

Comment:
Since this appears to crop up a lot, maybe we should have a helper method to encapsulate it? Something like `_try_convert_to_native`?

Reply:
+1